src/components/SeqVizParamsModal.vue

```diff
@@ -1,118 +1,271 @@
-                    </button>
-                    <button
-                        v-if="loadingInstance"
-                        v-on:click="$emit('submitInstance')"
-                        type="button"
-                        class="btn btn-primary">
-                        Add this sequence
-                    </button>
-                    <button
-                        v-on:click="$emit('closeModal')"
-                        type="button"
-                        class="btn btn-secondary"
-                        data-dismiss="modal">
-                        Close
-                    </button>
-                </div>
+                        <small
+                            v-bind:id="name + '-help-text'"
+                            class="form-text text-muted">
+                            {{ param.description }}
+                        </small>
+                    </div>
+                    <small v-if="Object.values(params).some(v => v.required)">
+                        <span class="redtext">*</span>Parameters with an
+                        asterisk are required.
+                    </small>
+                </form>
+            </div>
+            <div class="modal-footer">
+                <button
+                    v-if="!loadingInstance"
+                    v-on:click="$emit('submitParams')"
+                    type="button"
+                    :disabled="requiredMissing"
+                    class="btn btn-primary">
+                    Save changes
+                </button>
+                <button
+                    v-if="loadingInstance"
+                    v-on:click="$emit('submitInstance')"
+                    type="button"
+                    :disabled="requiredMissing"
+                    class="btn btn-primary">
+                    Add this sequence
+                </button>
+                <button
+                    v-on:click="$emit('closeModal')"
+                    type="button"
+                    class="btn btn-secondary"
+                    data-dismiss="modal">
+                    Close
+                </button>
             </div>
         </div>
     </div>
 </template>
 
 <script lang="ts">
     import Vue from 'vue'
-    import {ParamType} from '@/shared/ParamType'
+    import p5 from 'p5'
+    import {ParamInterface} from '@/shared/Paramable.ts'
 
     export default Vue.extend({
         name: 'SeqVizParamsModal',
-        methods: {},
+        methods: {
+            groupClass(indent: number): string {
+                let retval = 'form-group'
+                if (indent) {
+                    retval += ` indent${indent}`
+                }
+                return retval
+            },
+            setBigint(e: Event, p: {value: BigInt}) {
+                const target = e.target as HTMLInputElement
+                try {
+                    p.value = BigInt(target.value)
+                } catch {
+                    // Continue with old value
```

Comment:
I don't understand this. From what I understand `p.value` is already a `BigInt` and then in a try block you try to use the `BigInt()` constructor to make it a `BigInt`. You then have an empty catch block, so presumably if you can't convert `p.value` to `BigInt`, this method will still work? Why? Shouldn't a `setBigInt` do what it's supposed to do or error out?